engine/src/main/scala/cromwell/engine/backend/io/filesystem/gcs/GcsFileSystemProvider.scala

```diff
@@ -95,8 +95,7 @@ class GcsFileSystemProvider private (storageClient: Try[Storage], executionConte
     case gcsPath: NioGcsPath =>
       Try(withRetry(client.objects.get(gcsPath.bucket, gcsPath.objectName).execute)) recover {
         case ex: GoogleJsonResponseException
-          if ex.getStatusCode == 404 =>
-          if (!gcsPath.isDirectory) throw new FileNotFoundException(path.toString)
+          if ex.getStatusCode == 404 => if (!gcsPath.isDirectory) throw new FileNotFoundException(path.toString)
```

Comment:
Because if `&&` is in the guard, the `case` will only match if gcsPath is *not* a directory.
So if it *is* a directory it will not recover and the `get` below will throw an exception, which we don't want (if it's a directory we need to always consider that it exists - even if doesn't make sense for GCS, as a lot of Nio operations check for file/directory existence before doing something)